### PR TITLE
Split up oauth package into oauth and account (Account app scaffolding)

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
This PR takes care of splitting up the oauth package into the oauth package (token exchange and authorization) to  account app (authentication and account management), while retaining the config structure to stay conform to our compatibility commitment.

References #1422 

#### Changes
- Split up oauth app and account app
  - Extract authentication logic from `oauth` package to  new `account` package
  - Rename `oauth.js` entrypoint to `account.js`
  - Rename `oauth` webui folder to `account`
  - Rename static assets
  - Update module import statements
  - Update various configs accordingly
    - `package.json` module name mapper for jest
    - webpack config for module aliases (`@oauth` to `@account`)
    - eslint alias config
    - Retain stack config structure
  - Rename github commit topic from `oauth` to `account`
- Some related reducer fixes
- Updating go tests accordingly

#### Testing

I did manual testing and updated the unit tests.

##### Regressions

This is a pretty extensive PR so there might be issues introduced to numerous aspects:
- Authentication, login and logout
- Token exchange and authorization
- Broken redirects
- Broken views

#### Notes for Reviewers

Here's a little recap of the plan since this is going on for a while now:
- We want to introduce the "Account" app to be a dedicated app dealing with user authentication, profile settings, session management, client authorization management and other concerns revolving around the user account
- The `oauth` package will remain in place but shrinked down to merely authorization concerns (oauth flow, token exchange, client authorization)
- The Account app will not be an OAuth client (like the Console) but authorize API action via the session token. Session cookie authorization was added via [Enable authorization via session cookie by kschiffer · Pull Request #2564 · TheThingsNetwork/lorawan-stack · GitHub](https://github.com/TheThingsNetwork/lorawan-stack/pull/2564)
- The frontend of the `oauth` server will be reduced to merely the authorization screen (to authorize OAuth clients)
- Since we cannot break existing configuration, I decided to use the `is.oauth.*` configuration for **both** the `oauth` and `account` package.
  - This means that both web servers use the same `/oauth` base route by default
  - It is intended, however, to overwrite the default to use `/account` instead
  - On the frontend we'll also use the same entry point `account.js` for both frontends
  - We'll also use the same "branding" for both frontends, since we want the experience to unified through the "Account" app, meaning that the remaining authorization screen of the `oauth` server will be handled by the `account.js` endpoint as well.
- This PR is a necessary scaffolding to be able to rebase the already done frontend work of the Account app onto a working basis

Note that the target branch is again `feature/1422-account-app`, so this is not targeting master/develop and is preliminary work for upcoming account app related PRs. The point of this is again to reduce review diffs, chunking things down into bits that are scoped and easy to review until eventually merging `feature/1422-account-app` into `v3.10` when all parts have been merged into it.

It's necessary to update the config variables to mount this in development mode:
```
export export TTN_LW_IS_OAUTH_UI_JS_FILE="libs.bundle.js account.js"
```

Main reviewers: @bafonins @htdvisser 
@adriansmares please double-check whether these changes introduce any issues regarding federated auth.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: cmd/internal/shared/identityserver/config.go [hunk 1]
```diff
@@ -28,15 +28,17 @@ import (
...
 			TemplateData: webui.TemplateData{
 				SiteName:      "The Things Stack for LoRaWAN",
+				Title:         "Account",
 				Language:      "en",
 				CanonicalURL:  shared.DefaultOAuthPublicURL,
 				AssetsBaseURL: shared.DefaultAssetsBaseURL,
-				IconPrefix:    "oauth-",
-				CSSFiles:      []string{"oauth.css"},
-				JSFiles:       []string{"oauth.js"},
+				IconPrefix:    "account-",
```
Comment (htdvisser) [inline cmd/internal/shared/identityserver/config.go]: Changing this means that deployments with custom branding will have to copy/rename their custom branding files.

File: pkg/oauth/server.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 
 	echo "github.com/labstack/echo/v4"
 	"github.com/openshift/osin"
+	sess "go.thethings.network/lorawan-stack/v3/pkg/account/session"
```
Comment (htdvisser) [inline pkg/oauth/server.go]: Why is this import alias needed?

File: pkg/oauth/server.go [hunk 3]
```diff
@@ -223,18 +223,10 @@ func (s *server) RegisterRoutes(server *web.Server) {
...
 	csrfMiddleware := middleware.CSRF("_csrf", "/", s.config.CSRFAuthKey)
 
-	api := root.Group("/api", csrfMiddleware)
-	api.POST("/auth/login", s.Login)
-	api.POST("/auth/logout", s.Logout, s.requireLogin)
-	api.GET("/me", s.CurrentUser, s.requireLogin)
-
 	page := root.Group("", csrfMiddleware)
-	page.GET("/login", webui.Template.Handler, s.redirectToNext)
 	page.GET("/logout", s.ClientLogout)
```
Comment (htdvisser) [inline pkg/oauth/server.go:230]: Maybe add a note that this route is in use by existing OAuth clients, meaning that it's public API and should therefore not be removed in v3.

File: pkg/oauth/server.go [hunk 4]
```diff
@@ -223,18 +223,10 @@ func (s *server) RegisterRoutes(server *web.Server) {
...
 	page.GET("/authorize", s.Authorize(webui.Template.Handler), s.redirectToLogin)
 	page.POST("/authorize", s.Authorize(webui.Template.Handler), s.redirectToLogin)
-	page.GET("/", webui.Template.Handler, s.redirectToLogin)
-	page.GET("/*", webui.Template.Handler)
```
Comment (htdvisser) [inline pkg/oauth/server.go:237]: I think we (used to) send emails linking to `/oauth`, so if we remove this, we're breaking those links, so let's keep the redirect

File: pkg/account/user.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+		return err
+	}
+	session, err := s.store.CreateSession(ctx, &ttnpb.UserSession{
+		UserIdentifiers: userIDs,
+		SessionSecret:   hashedSecret,
+	})
+	if err != nil {
+		return err
+	}
+	events.Publish(evtUserLogout.NewWithIdentifiersAndData(ctx, userIDs, nil))
```
Comment (htdvisser) [inline pkg/account/user.go:97]: This doesn't look like the place to send a `evtUserLogout`

File: pkg/account/session/observability.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		events.WithVisibility(ttnpb.RIGHT_USER_ALL),
+		events.WithAuthFromContext(),
+		events.WithClientInfoFromContext(),
+	)
+	evtUserLogout = events.Define(
+		"account.user.logout", "logout user",
+		events.WithVisibility(ttnpb.RIGHT_USER_ALL),
+		events.WithAuthFromContext(),
+		events.WithClientInfoFromContext(),
+	)
```
Comment (htdvisser) [inline pkg/account/session/observability.go]: These do not seem to be used in this package

File: pkg/account/observability.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		events.WithVisibility(ttnpb.RIGHT_USER_ALL),
+		events.WithAuthFromContext(),
+		events.WithClientInfoFromContext(),
+	)
+	evtUserLoginFailed = events.Define(
+		"oauth.user.login_failed", "login user failure",
+		events.WithVisibility(ttnpb.RIGHT_USER_ALL),
+		events.WithAuthFromContext(),
+		events.WithClientInfoFromContext(),
+	)
```
Comment (htdvisser) [inline pkg/account/observability.go]: This doesn't seem to be used in this package.

**Review by htdvisser — COMMENTED** [review]
Did a quick initial review. Will take a closer look another time.

**Comment by johanstokking** [comment]
> Here's a little recap of the plan since this is going on for a while now:
> [...]

Totally agreed with this.

**Review by adriansmares — COMMENTED** [review]
Took a look at the changes to `pkg/oauth` and I consider this to be compatible with the federated auth (which will probably migrate to `pkg/account`).

**Comment by htdvisser** [comment]
Nothing to add to my previous review

File: pkg/oauth/server.go (hunk 4 above)
Comment (kschiffer) [reply pkg/oauth/server.go:237]: But the route will be handled by the account server. Am I missing something here?

Comment (htdvisser) [reply pkg/oauth/server.go:237]: Okay, then it's fine. I thought the account server was mounted somewhere else.

**Review by htdvisser — APPROVED** [review]
Go stuff LGTM

**Pull request merged**